starknet_devnet/blocks.py

```diff
@@ -99,85 +99,40 @@ async def generate(
         """
         block_number = self.get_number_of_blocks()
         timestamp = state.state.block_info.block_timestamp
-        signature = transaction.get_signature()
-
         if block_number == 0:
             parent_block_hash = 0
         else:
             last_block = self.__get_last_block()
             parent_block_hash = last_block.block_hash
 
-        if self.lite:
-            block_hash = block_number
+        if is_empty_block:
+            transaction_receipts = ()
+            transactions = []
         else:
```

Comment:
This `else` will be run even in lite mode (basically every time that `is_empty_block` is not `True`), which is not the previous behavior.

Reply:
aah i see, should have used the the more explicit elif then

what do you think?

Reply:
not sure if just replacing `else` with `elif CONDITION` would be enough

Reply:
i have pushed it, kindly check if you have the time @FabijanC

thank you!

Reply:
The point of lite mode is to avoid unnecessary calculation, like calculating block hash. This wasn't handled properly in the code which I commented on and the calculation was done even in lite mode. So functionally, the behavior was the same, but not performance-wise (it was slower). But it looks better after the changes, I'll quickly do a formal review.

Reply:
yes i agree, performance wise, in the commit before this, it was slower because of the reason you have pointed!

but now i think the latest commit resolves this issue.

thanks for your review!